packages/react-celo/src/utils/useTheme.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react';
+
+import { useReactCeloContext } from '../react-celo-provider';
+import defaultTheme from '../theme/default';
+import { Mode, Theme } from '../types';
+
+class ModeState {
+  private _mode = Mode.Light;
+  private listeners = [] as ((mode: Mode) => void)[];
+
+  public set mode(mode: Mode) {
+    if (this._mode === mode) return;
+
+    this._mode = mode;
+    this.listeners.forEach((cb) => cb(this._mode));
+  }
+
+  public get mode() {
+    return this._mode;
+  }
+
+  public addEventListener(listener: (mode: Mode) => void) {
+    this.listeners.push(listener);
+  }
+  public removeEventListener(listener: (mode: Mode) => void) {
+    this.listeners = this.listeners.filter((x) => x !== listener);
+  }
+  public removeEventListeners() {
+    this.listeners = [];
+  }
+}
+const state = new ModeState();
+
+// const COLOR_SCHEME_QUERY = '(prefers-color-scheme: dark)';
+const DARK_CLASSES = ['dark', 'tw-dark'];
+function hasDarkClasses(elem: HTMLElement) {
+  // const hasDarkMedia = window.matchMedia(COLOR_SCHEME_QUERY).matches;
+  return DARK_CLASSES.some((cls) => elem.classList.contains(cls));
+}
+
+if (typeof MutationObserver !== 'undefined') {
+  const body = document.getElementsByTagName('body');
+  const html = document.getElementsByTagName('html');
+  const elems = [...body, ...html];
+
+  const mutationCallback = () => {
+    const isDark = elems.some(hasDarkClasses);
+    state.mode = isDark ? Mode.Dark : Mode.Light;
+  };
+
+  const observer = new MutationObserver(mutationCallback);
+  mutationCallback();
+  elems.forEach((elem) => {
+    observer.observe(elem, {
+      subtree: false,
+      childList: false,
+      attributeFilter: ['class'],
+    });
+  });
+}
+
+export default function useTheme(): Theme {
+  const [{ theme }] = useReactCeloContext();
+  const [mode, setMode] = useState(state.mode);
+
+  useEffect(() => {
+    const cb = (m: Mode) => setMode(m);
+    state.addEventListener(cb);
+    return () => {
+      state.removeEventListener(cb);
+    };
+  }, []);
+
+  return theme || defaultTheme[mode];
+}
```

Comment:
Note: That hook could be soooo much simpler if we didn't handle the html or body dark or tw-dark classnames.